Sources/Plasma/PubUtilLib/plPipeline/plTextFont.cpp

```diff
@@ -104,183 +98,137 @@ uint16_t  *plTextFont::IInitFontTexture()
-                *tBits = 0xffff;
-            else
-                *tBits = 0;
-
-            tBits++;
-        }
-    }
+    SelectObject(hDC, hFont);
+    
+    DWORD fontDataSize = GetFontData( hDC, 0, 0, 0, 0 );
+    void* fontData = std::malloc( fontDataSize );
+    GetFontData(hDC, 0, 0, fontData, fontDataSize);
+    ftError = FT_New_Memory_Face(library, (FT_Byte *) fontData, fontDataSize, 0, &face);
+    FT_UInt freeTypeResolution = GetDeviceCaps(hDC, LOGPIXELSY);
 
-    // Cleanup and return
-    DeleteObject( hBitmap );
     DeleteDC( hDC );
     DeleteObject( hFont );
 
-    return data;
 #elif defined(HS_BUILD_FOR_APPLE)
-    int     nHeight, x, y, c;
-    char    myChar[ 2 ] = "x";
-    uint16_t  *tBits;
-
-    uint32_t       *bitmapBits;
-    uint32_t        bAlpha;
-
     
-    // Figure out our texture size
-    if( fSize * 2 > 40 )
-        fTextureWidth = fTextureHeight = 1024;
-    else if( fSize * 2 > 20 )
-        fTextureWidth = fTextureHeight = 512;
-    else
-        fTextureWidth = fTextureHeight = 256;
-
-
-    // Create a new DC and bitmap that we can draw characters to
-    CGColorSpaceRef colorSpace = CGColorSpaceCreateWithName(kCGColorSpaceGenericRGB);
-    CGContextRef bitmapContext = CGBitmapContextCreate(nullptr,
-                                                fTextureWidth,
-                                                fTextureHeight,
-                                                8,
-                                                4 * fTextureWidth,
-                                                colorSpace,
-                                                kCGImageAlphaPremultipliedLast);
-    CFRelease(colorSpace);
-    CGContextSetTextMatrix(bitmapContext, CGAffineTransformMakeScale(1, -1));
-    CGContextTranslateCTM(bitmapContext, 0, fTextureHeight);
-    CGContextScaleCTM(bitmapContext, 1, -1);
-    CGContextScaleCTM(bitmapContext, 2.0, 2.0);
+    CFStringRef fontName = CFStringCreateWithCString(nullptr, fFace, kCFStringEncodingUTF8);
+    CTFontDescriptorRef fontDescriptor = CTFontDescriptorCreateWithNameAndSize( fontName, 0.0f );
+    CTFontDescriptorRef fulfilledFontDescriptor = CTFontDescriptorCreateMatchingFontDescriptor(fontDescriptor, nullptr);
+    hsAssert(fulfilledFontDescriptor != nullptr, "Cannot create Mac font");
+    CFRelease(fontName);
+    CFRelease(fontDescriptor);
+    
+    CFURLRef fontURL = (CFURLRef) CTFontDescriptorCopyAttribute(fulfilledFontDescriptor, kCTFontURLAttribute);
+    CFStringRef fileSystemPath = CFURLCopyFileSystemPath(fontURL, kCFURLPOSIXPathStyle);
+    char cPath[PATH_MAX];
+    CFStringGetCString(fileSystemPath, cPath, PATH_MAX, kCFStringEncodingUTF8);
+    
+    ftError = FT_New_Face(library, cPath, 0, &face);
     
-    bitmapBits = (uint32_t *)CGBitmapContextGetData(bitmapContext);
+    CFRelease(fulfilledFontDescriptor);
+    CFRelease(fontURL);
+    CFRelease(fileSystemPath);
+    
+    FT_UInt freeTypeResolution = 192;
```

Comment:
This is a giant ugly hack. I need to come up with something to get the DPI through in since we don't have access to Cocoa in here.